server/game/gameSystems/LookMoverDeckCardsTopOrBottomSystem.ts

```diff
@@ -23,59 +20,67 @@ export class LookMoveDeckCardsTopOrBottomSystem<TContext extends AbilityContext
+            const actual_amount = Math.min(amount, deckLength);
+            const cards = player.drawDeck.slice(0, actual_amount);
 
             // @ts-ignore
-            const choiceHandler = (player) => {
-                if (cards.length > 0) {
-                    context.game.promptWithHandlerMenu(player, {
-                        activePromptTitle: 'Select card to move to the top or bottom of the deck',
-                        choices: cards.map((card: Card) => [
-                            'Put ' + card.internalName + ' on top',
-                            'Put ' + card.internalName + ' to bottom',
-                        ]).flat(),
-                        handlers: cards.map((card: Card) => [(() => {
-                            events.push(moveCard({
-                                target: card,
-                                bottom: false,
-                                destination: Location.Deck
-                            }).generateEvent(context.target, context));
-                            // get rid of the card from cards
-                            cards = cards.filter((a) => a !== card);
-                            choiceHandler(context.player);
-                        }),
-                        (() => {
-                            events.push(moveCard({
-                                target: card,
-                                bottom: true,
-                                destination: Location.Deck
-                            }).generateEvent(context.target, context));
-                            // get rid of the card from cards
-                            cards = cards.filter((a) => a !== card);
-                            choiceHandler(context.player);
-                        })]).flat()
-                    });
-                } else {
+            // Each card has two options to be put on top or on bottom for each option we have a handler whcih
+            // recursively calls the function and removes handlers from the list until the card pool reaches 0.
+            const choiceHandler = (player, cards: any[]) => {
+                if (cards.length === 0) {
                     return true;
                 }
-            };
+                // setup the choices for each card top and bottom
+                const handlerChoices = cards.map((card: Card) => [
+                    'Put ' + card.title + ' on top',
+                    'Put ' + card.title + ' on bottom',
+                ]).flat();
 
-            choiceHandler(context.player);
+                context.game.promptWithHandlerMenu(player, {
+                    activePromptTitle: 'Select card to move to the top or bottom of the deck',
+                    choices: handlerChoices,
+                    handlers: cards.map((card: Card) => [(() => {
+                        this.pushMoveEvent(false, card, cards, events, context, choiceHandler);
+                    }),
+                    (() => {
```

Comment:
Since each call is just one statement now you can remove the `{ }` blocks and effectively just do `() => this.pushMoveEvent(...)`